drivers/ipm/Kconfig.ivshmem

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2023, Linaro
+
+if IPM_IVSHMEM
+
+config IPM_IVSHMEM_EVENT_LOOP_STACK_SIZE
+	int "Stack size in bytes of IVSHMEM IPM Event loop task"
+	default 8192
+	help
+	  Adjust the stack size, in bytes of the ivshmem event loop task.
+
+config IPM_IVSHMEM_EVENT_LOOP_PRIO
+	int "Priority of IVSHMEM IPM Event loop task"
+	default 2
```

Comment:
Why do you need such high cooperative priority? Cooperative tasks must really be carefully justified because can easily starve the scheduler if used not appropriately.

Reply:
It is just a way to emulate the cross-core interrupt service, as other IPM drivers (or MBOX) conventions, heavy work should not be done there.

Reply:
The problem is that in the cooperative thread you are going to call `dev_data->cb`, that means that the user is basically installing an handler in a very high priority cooperative thread without even knowing it.

Cooperative threads are dangerous because if the user doesn't know what he is doing or even that the code that he installed is in a cooperative threads, things can go south very quickly.

Think about someone installing in the callback a busy-loop of some kind, that means that the scheduler is basically hostage of the user that is DoS-sing the whole platform.

As someone said: if you feel the need to use a cooperative thread, you are probably wrong.

Reply:
I understand the implication of usage the cooperative threads without caution, my point is more from how the user perceiveis the consumption of the IPM driver, and it is documented as follow:

Reply:
I see but I still do not see why a cooperative thread must be used here. Yes, you are more compliant to the API but at what cost? You are offloading too much responsibility to the user IMO for what gain? I can understand the usage of cooperative threads when you really, really want to be sure to not be interrupted, but if you do not have that kind of need or some kind of performance constrain the usage of cooperative threads is to be avoided as much as possible.

Please note that I'm suggesting this because I was personally heavily bitten in the past by the unnecessarily usage of cooperative threads, see https://github.com/zephyrproject-rtos/zephyr/pull/46172